src/docs-rs/developing.md

```diff
@@ -0,0 +1,283 @@
+
+```console
+$ mkdir -p ignored/cratesfyi-prefix
+$ cd ignored/cratesfyi-prefix
+$ mkdir -vp documentations public_html sources
+$ git clone https://github.com/rust-lang/crates.io-index && cd crates.io-index
+$ git branch crates-index-diff_last-seen
+```
+
+(That last command is used to set up the `crates-index-diff` crate, so we can start monitoring new crate releases.)
+
+## Docker group
+
+docs.rs needs to run docker containers for sandboxing. Therefore, you need to be in the 'docker' group to build crates. If you are already in the docker group, you can skip this step (you can check with `groups`).
+
+```console
+# usermod -a -G docker "$USER"
+$ # now logout and back in to your shell
+$ cd /path/to/docs.rs/ignored/cratesfyi-prefix
+```
+
+## Environment for the server
+
+To ensure that the docs.rs server is configured properly, we need to set a few environment variables. This is most easily done by making a `.env` file in the crate root.
+The `.env.sample` file already exists in the repo with some helpful defaults, but you may want to make adjustments based off of your setup.
+
+```sh
+cd ..
+cp .env.sample .env
+```
+
+The variables that are most commonly used are `CRATESFYI_PREFIX`, which gives the root url of the server and `CRATESFYI_DATABASE_URL` which is the location url of the postgres database that
+docs.rs will attempt to connect with.
+
+```console
+$ cargo run database migrate
+$ cargo run database update-search-index
+$ cargo run database update-release-activity
+# This will take between 5 and 30 minutes on the first run, depending on your internet speed.
+# It downloads the rustops/crates-build-env crates which is almost 1 GB.
+# It does not currently display a progress bar, this is https://github.com/rust-lang/rustwide/issues/9
+# As a workaround, you can run `docker pull rustops/crates-build-env` in a separate terminal.
+$ cargo run build crate rand 0.5.5
+```
+
+## Developing with Windows
+
+### Requirements
+
+* [Rust](https://rustup.rs/)
+* [Git](https://gitforwindows.org/)
+* [Docker](https://www.docker.com/products/docker-desktop)
+* [docker-compose](https://docs.docker.com/compose/install/)
+
+Note: If you don't want to use docker and docker-compose, it is possible to manually install Postgres and use that for running tests, but the docker route is recommended.
+
+### Quickstart
+
+Make sure you are able to download ~10GiB of data and are willing to wait a while, the build and download takes a while.
+
+```powershell
+git clone https://github.com/rust-lang/docs.rs.git docs.rs
+cd docs.rs
+Copy-Item -Path .env.sample -Destination .env
+
+docker-compose build
+```
+
+Next for testing you'll need to run the database so that tests can connect to it.
+
+```powershell
+# To run only the database
+docker-compose run -p 5432:5432 -d db
+
+# To run the database, s3 and the webserver
+docker-compose up
```

Comment:
I don't understand the point of this section. All the steps are almost identical to the linux version, the only difference is `Copy-Item` which I'm pretty sure is aliased to `cp` by default anyway.